src/client/extensionActivation.ts

```diff
@@ -86,9 +87,11 @@ async function activateLegacy(
     // Load the two data science experiments that we need to register types
     // Await here to keep the register method sync
     const experimentService = serviceContainer.get<IExperimentService>(IExperimentService);
+    const amlCompute = serviceContainer.get<AmlComputeContext>(AmlComputeContext);
     experimentService.logExperiments();
+
     let useVSCodeNotebookAPI =
-        env.appName.includes('Insider') || (await experimentService.inExperiment(Experiments.NativeNotebook));
+        amlCompute.isAmlCompute || (await experimentService.inExperiment(Experiments.NativeNotebook));
```

Comment:
I think we still need the 'in insiders' check, right? Or is that handled elsewhere now?

Reply:
That's handled in experiment class (was always there, hence removed that check from this place)